www/FileWriter.js

```diff
@@ -107,8 +108,29 @@ FileWriter.prototype.write = function(data) {
         var fileReader = new FileReader();
         fileReader.onload = function() {
             // Call this method again, with the arraybuffer as argument
-            FileWriter.prototype.write.call(that, this.result);
+            FileWriter.prototype.write.call(that, this.result, true /* isPendingBlobReadResult */);
         };
+        fileReader.onerror = function (evt) {
+            // DONE state
+            that.readyState = FileWriter.DONE;
+
+            // Save error
+            that.error = evt.target.error;
```

Comment:
I think **evt.target** here is the same as **this** in the callback, and they both refer to the FileReader instance, right? It's a bit confusing that you use **evt.target** here, but **this** is used in the **onload** callback. (I can't actually find documentation that says **this** in a FileReader callback is set to the FileReader, though it seems like a reasonable assumption.)